wicker/core/storage.py

```diff
@@ -79,9 +147,29 @@ def fetch_file(self, input_path: str, local_prefix: str, timeout_seconds: int =
         :param timeout_seconds: number of seconds till timing out on waiting for the file to be downloaded
         :return: local path to the file on the local file system
         """
-        input_path = os.path.join(input_path, os.path.basename(local_prefix))
-        self.download_with_retries(input_path, local_prefix)
-        return local_prefix
+        local_full_path = os.path.join(local_prefix, os.path.basename(input_path))
+        return self._safe_file_download(
```

Comment:
@pickles-bread-and-butter I want to make a change to the `FileSystemDataStorage` routine `fetch_file`. All I want to do in this function is to calculate the `local_full_path` and call `shutil.copyfile`. I don't want to use retries or to create a lock file for local storage.

I'm concerned we are going to do something to wreck the FUSE storage performance. I don't think we should add retries or lock files for local storage until we determine that there is actually a reliability problem that we need to solve. I'm sorry I didn't catch this before.

For now, I think you should remove the `download_with_retries` and `_safe_file_download` functions about of the `AbstractDataStorage` base class and push them back into the `S3DataStorage` class.

This will also enable you to restore the original function signature `def download_with_retries(self, bucket: str, key: str, local_path: str)` in the `S3DataStorage class` (which was a backwards-incompatible change).

Again, apologies for wanting to make this change late in the review process for this PR!

Reply:
So what I did was, I kept the `_safe_file_download` in the base and added another head on top of it `_download`. That is an abstract method a user will need to implement. Then I removed the retries and one unneeded lock from `_safe_file_download`. I moved the `download_with_retries` back into the `S3Dataset` class and called it through the new `_download` head to make hopeful eventual deprecation possible. Let me know if that addresses your concerns.

Reply:
Offline discussion - I ended up reverting the download_with_retries to the original in the S3Dataset class. Then in the File System one I used the same logic almost but without the retry loop (probably causing hangs in the download).